src/test/kotlin/no/nav/tilleggsstonader/sak/brev/BrevServiceTest.kt

```diff
@@ -0,0 +1,199 @@
+package no.nav.tilleggsstonader.sak.brev
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import no.nav.tilleggsstonader.sak.behandling.domain.BehandlingStatus
+import no.nav.tilleggsstonader.sak.behandlingsflyt.StegType
+import no.nav.tilleggsstonader.sak.brev.BrevService.Companion.BESLUTTER_SIGNATUR_PLACEHOLDER
+import no.nav.tilleggsstonader.sak.fagsak.domain.PersonIdent
+import no.nav.tilleggsstonader.sak.infrastruktur.database.Fil
+import no.nav.tilleggsstonader.sak.infrastruktur.database.SporbarUtils
+import no.nav.tilleggsstonader.sak.infrastruktur.database.repository.findByIdOrThrow
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.Feil
+import no.nav.tilleggsstonader.sak.infrastruktur.sikkerhet.SikkerhetContext
+import no.nav.tilleggsstonader.sak.util.BrukerContextUtil.clearBrukerContext
+import no.nav.tilleggsstonader.sak.util.BrukerContextUtil.mockBrukerContext
+import no.nav.tilleggsstonader.sak.util.behandling
+import no.nav.tilleggsstonader.sak.util.fagsak
+import no.nav.tilleggsstonader.sak.util.saksbehandling
+import org.assertj.core.api.AssertionsForClassTypes.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.http.HttpStatus.BAD_REQUEST
+
+internal class BrevServiceTest {
+
+    private val fagsak = fagsak(setOf(PersonIdent("12345678910")))
+    private val behandling = behandling(fagsak)
+
+    private val vedtaksbrevRepository = mockk<VedtaksbrevRepository>()
+    private val familieDokumentClient = mockk<FamilieDokumentClient>()
+
+    private val brevService = BrevService(vedtaksbrevRepository, familieDokumentClient)
+
+    private val vedtaksbrev: Vedtaksbrev = lagVedtaksbrev("malnavn")
+    private val beslutterNavn = "456"
+
+    @BeforeEach
+    fun setUp() {
+        mockBrukerContext(beslutterNavn)
+    }
+
+    @AfterEach
+    fun tearDown() {
+        clearBrukerContext()
+    }
+
+    @Test
+    internal fun `lagBeslutterBrev - skal kaste feil hvis behandlingen ikke har riktig steg`() {
+        every { vedtaksbrevRepository.findByIdOrThrow(any()) } returns vedtaksbrev
+
+        val feil = assertThrows<Feil> {
```

Comment:
I stedet for en blandning av asserts fra junit og assertj så kan vi kanskje bruke `catchThrowableOfType` 
Ser at jeg også tatt inn denne assertThrows noen steder :/

Reply:
Fikser 💯